src/Raven.Server/Documents/Indexes/Static/JavaScriptMapOperation.cs

```diff
@@ -55,8 +55,14 @@ public IEnumerable IndexingFunction(IEnumerable<object> items)
                         }
                         catch (StatementsCountOverflowException e)
                         {
-                            throw new Raven.Client.Exceptions.Documents.Patching.JavaScriptException(
-                                $"The maximum number of statements executed has been reached. You can configure it by modifying the configuration option: '{RavenConfiguration.GetKey(x => x.Indexing.MaxStepsForScript)}'.", e);
+                            throw new Client.Exceptions.Documents.Patching.JavaScriptException(
+                                $"The maximum number of statements executed has been reached. You can configure it by modifying the configuration option: '{RavenConfiguration.GetKey(x => x.Indexing.MaxStepsForScript)}'.",
+                                e);
+                        }
+                        catch (JavaScriptException jse) when (jse.Message.Contains("String compilation has been disabled in engine options"))
+                        {
+                            throw new Client.Exceptions.Documents.Patching.JavaScriptException(
```

Comment:
don't we want a test that actually checks that we get this exception when String compilation is disabled in engine options

Reply:
+1

Reply:
@grisha-kotler please send a dedicated PR with that. I'm merging this one.